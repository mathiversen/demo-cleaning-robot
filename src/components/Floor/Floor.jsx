import React from 'react'
import { css } from 'goober'

export default ({ children, columns, rows }) => {
    const classname = css`
        display: grid;
        grid-template-columns: repeat(${columns}, 1fr);
        grid-template-rows: repeat(${rows}, 1fr);
        grid-gap: 2px;
        background: black;
        border: solid 2px black;
    `
    return <section className={classname}>{children}</section>
}
