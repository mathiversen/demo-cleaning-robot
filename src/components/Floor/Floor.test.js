import React from 'react'
import Floor from './Floor'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
    const mockProps = {
        columns: 10,
        rows: 10,
        children: [<div key="x">child</div>]
    }
    const tree = renderer.create(<Floor {...mockProps} />).toJSON()
    expect(tree).toMatchSnapshot()
})
