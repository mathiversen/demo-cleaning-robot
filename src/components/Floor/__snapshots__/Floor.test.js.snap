// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="go1201451965"
>
  <div>
    child
  </div>
</section>
`;
